src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -199,6 +199,26 @@ public function print_client_interactivity_data() {
 		_deprecated_function( __METHOD__, '6.7.0' );
 	}
 
+	/**
+	 * Set client-side interactivity-router data.
+	 *
+	 * Once in the browser, the state will be parsed and used to hydrate the client-side
+	 * interactivity stores and the configuration will be available using a `getConfig` utility.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param array $data Data to filter.
+	 * @return array Data for the Interactivity Router script module.
+	 */
+	public function filter_script_module_interactivity_router_data( array $data ): array {
```

Comment:
Can't we find a better way?

This is really a step backwards from the JS I18N mechanism we implemented all those years ago 😞 

Translators loose important context because the strings are no longer co-located with the code where they're being used. This also fails as soon as soon as you are dealing with plurals.

Reply:
This comment and conversation continued in https://github.com/WordPress/gutenberg/pull/65123#discussion_r1754292031.